### PR TITLE
analyzer: Also try to continue if building a Maven project from file fails

### PR DESCRIPTION
**Pull request by mnonnenmacher**
Similar to 5c7d98e continue if a ProjectBuildingException occurs while
building a project from a POM file, if the exception contains a
ProjectBuildingResult.

---
This change is [Reviewable](https://reviewable.io/reviews/heremaps/oss-review-toolkit/622)

### CODE REVIEW

File: analyzer/src/main/kotlin/MavenSupport.kt [hunk 1]
```diff
@@ -329,7 +348,7 @@ class MavenSupport(localRepositoryManagerConverter: (LocalRepositoryManager) ->
             } catch (e: ProjectBuildingException) {
                 e.showStackTrace()
 
-                val failedProject = e.results.find { projectBuildingResult ->
+                val failedProject = e.results?.find { projectBuildingResult ->
```
Comment (sschuberth) [inline analyzer/src/main/kotlin/MavenSupport.kt]: Should this, strictly speaking, be a separate commit as it seems to be a fixup for 5c7d98e2c9e25713dbef3eb1e1461c01eccf1f4f?

Comment (mnonnenmacher) [reply analyzer/src/main/kotlin/MavenSupport.kt]: Done.
